src/allmydata/storage_client.py

```diff
@@ -124,7 +143,25 @@ def get_servers_for_psi(self, peer_selection_index):
         def _permuted(server):
             seed = server.get_permutation_seed()
             return sha1(peer_selection_index + seed).digest()
-        return sorted(self.get_connected_servers(), key=_permuted)
+        try:
+            import string
+            hook_str = self.server_selection_hook
+            module_set = hook_str.split('.')
+
+            # somehow we need import the whole module first
+            # duplicate a list first for del operation
+            module_only = module_set[:-1]
+            __import__(string.join(module_only, '.'))
+
+            # iterate the module path to find the hook function
+            m = __import__(module_set[0])
+            module_set = module_set[1:]
+            for module_lvl_name in module_set:
+                m = getattr(m, module_lvl_name)
+            return m(self.get_connected_servers(), key=_permuted)
+        except:
+            # Explicitly catch the wrong configuration in tahoe.cfg or 3rd party module is not installed case.
```

Comment:
A bare "except" is bad. The most-general should be `except Exception:` but ideally something more-specific (i.e. catch a more-specific `Exception` subclass). Also, I would guard the "import stuff" separately from the "invoke the hook" part. Besides all that, though, this particular `try/except` doesn't actually do anything and can be removed.